l10n_fr_siret/partner.py

```diff
@@ -58,6 +57,12 @@ def _get_siret(self):
     @api.constrains('siren', 'nic')
     def _check_siret(self):
         """Check the SIREN's and NIC's keys (last digits)"""
+        # Check unicity of siret
+        for partner in self.env['res.partner'].search([('id', '!=', self.id),('siret', 'not in', [False,''])]):
+            siret = str(self.siren) + str(self.nic)
+            if siret == partner.siret:
+                raise ValidationError(_("Siret must be unique !\n It was defined for '%s'.") % partner.name)
+
```

Comment:
@gurneyalex you're right it could be better to use UNIQUE SQL index on big database .. will try it and update code accordingly.